### DISCUSSION TITLE
Retain the form data in the fields

### DISCUSSION THREAD

**Discussion by sivaputrevu** [react-hook-form/react-hook-form#2590] [General]
Hi,

I am using react hook form in my react-redux application. my form resides inside a MUI dialog.
once the user fills the mandatory fields and clicks on submit the call goes to redux layer and invokes API and then the control comes back to the UI form. if the calls is success I am closing dialog window as the call success, if there is an error how to show the previously entered user data in the form. currently it is showing till some clicks on submit[lets say 20+ times] and after that the dialog is closing.
I tried to disable the submit button if there are server errors but not able to handle it effectively.
any thoughts?

thanks

**Comment by bluebill1049** [comment]
`shouldUnregister: false`
